src/issues/github.rs

```diff
@@ -1,44 +1,79 @@
-use color_eyre::eyre::{eyre, Result};
+use color_eyre::eyre::{eyre, Result, WrapErr};
 
-use crate::app_config::get_or_prompt_for_octocrab;
+use crate::app_config::get_or_prompt_for_github_token;
 use crate::issues::Issue;
 use crate::{config, state};
 
-pub(crate) async fn list_issues(
+const ISSUES_QUERY: &str = r##"
+query($repo: String!, $owner: String!, $labels: [String!]) { 
+  repository(name:$repo, owner:$owner) { 
+    issues(states:OPEN, first: 30, labels: $labels) {
+      nodes {
+        number,
+        title
+      }
+    }
+  }
+}
+"##;
+
+#[derive(serde::Deserialize)]
+struct ResponseIssue {
+    number: usize,
+    title: String,
+}
+
+pub(crate) fn list_issues(
     github_config: Option<config::GitHub>,
     github_state: state::GitHub,
-    labels: Option<Vec<String>>,
+    labels: &Option<Vec<String>>,
 ) -> Result<(Option<config::GitHub>, state::GitHub, Vec<Issue>)> {
     match github_config {
         None => Err(eyre!("GitHub is not configured")),
         Some(github_config) => {
-            let octocrab = match github_state {
-                state::GitHub::Initialized { octocrab } => octocrab,
-                state::GitHub::New => get_or_prompt_for_octocrab()?,
-            };
-            let request = octocrab.issues(&github_config.owner, &github_config.repo);
-            let oc_issues = if let Some(labels) = labels {
-                request.list().labels(labels.as_slice()).send().await?
-            } else {
-                request.list().send().await?
+            let token = match github_state {
+                state::GitHub::Initialized { token } => token,
+                state::GitHub::New => get_or_prompt_for_github_token()?,
             };
-            let issues = oc_issues
-                .items
+            let response = ureq::post("https://api.github.com/graphql")
+                .set("Authorization", &format!("bearer {}", &token))
```

Comment:
Nitpick: I don't think the reference in `&token` is necessary in a macro like `format!`.